cmd/juju/action/listoperations.go

```diff
@@ -197,58 +200,142 @@ func operationDisplayTime(r params.ActionResult) time.Time {
+	Status  string                 `yaml:"status" json:"status"`
+	Action  *actionSummary         `yaml:"action,omitempty" json:"action,omitempty"`
+	Timing  timingInfo             `yaml:"timing,omitempty" json:"timing,omitempty"`
+	Tasks   map[string]taskSummary `yaml:"tasks,omitempty" json:"tasks,omitempty"`
 }
 
-type byId []params.ActionResult
+type timingInfo struct {
+	Enqueued  string `yaml:"enqueued,omitempty" json:"enqueued,omitempty"`
+	Started   string `yaml:"started,omitempty" json:"started,omitempty"`
+	Completed string `yaml:"completed,omitempty" json:"completed,omitempty"`
+}
 
-func (s byId) Len() int {
-	return len(s)
+type actionSummary struct {
+	Name       string                 `yaml:"name" json:"name"`
+	Parameters map[string]interface{} `yaml:"parameters" json:"parameters"`
 }
-func (s byId) Swap(i, j int) {
-	s[i], s[j] = s[j], s[i]
+
+type taskSummary struct {
+	Name   string     `yaml:"name,omitempty" json:"name,omitempty"`
+	Host   string     `yaml:"host" json:"host"`
+	Status string     `yaml:"status" json:"status"`
+	Timing timingInfo `yaml:"timing,omitempty" json:"timing,omitempty"`
 }
-func (s byId) Less(i, j int) bool {
-	if s[i].Action == nil {
-		return true
+
+// formatOperationResult inserts the remaining ones in a map[string]interface{} for cmd.Output to
+// write in an easy-to-read format.
+func formatOperationResult(operation params.OperationResult, utc bool) operationSummary {
+	result := operationSummary{
+		Summary: operation.Summary,
+		Status:  operation.Status,
+		Timing: timingInfo{
+			Enqueued:  formatTimestamp(operation.Enqueued, false, utc, false),
+			Started:   formatTimestamp(operation.Started, false, utc, false),
+			Completed: formatTimestamp(operation.Completed, false, utc, false),
+		},
+		Tasks: make(map[string]taskSummary, len(operation.Actions)),
+	}
+	var singleAction actionSummary
+	haveSingleAction := true
+	for i, task := range operation.Actions {
+		if task.Action == nil {
+			result.Status = "error"
+			continue
+		}
+		tag, err := names.ParseActionTag(task.Action.Tag)
+		if err != nil {
+			// Not expected to happen.
+			continue
+		}
+		taskInfo := taskSummary{
+			Host:   task.Action.Receiver,
+			Status: task.Status,
+			Timing: timingInfo{
+				Enqueued:  formatTimestamp(task.Enqueued, false, utc, false),
+				Started:   formatTimestamp(task.Started, false, utc, false),
+				Completed: formatTimestamp(task.Completed, false, utc, false),
+			},
+		}
+		ut, err := names.ParseUnitTag(task.Action.Receiver)
+		if err == nil {
+			taskInfo.Host = ut.Id()
+		}
+		result.Tasks[tag.Id()] = taskInfo
+		if i == 0 {
+			singleAction = actionSummary{
+				Name:       task.Action.Name,
+				Parameters: make(map[string]interface{}),
+			}
+			for k, v := range task.Action.Parameters {
+				singleAction.Parameters[k] = v
+			}
+			continue
+		}
+		if haveSingleAction && (task.Action.Name != singleAction.Name || !reflect.DeepEqual(task.Action.Parameters, singleAction.Parameters)) {
```

Comment:
break this condition out into a variable that has a name that explains this a little better.

Reply:
added code comment as well